include/drivers/ipm.h

```diff
@@ -221,6 +232,34 @@ static inline int z_impl_ipm_set_enabled(const struct device *ipmdev,
 	return api->set_enabled(ipmdev, enable);
 }
 
+/**
+ * @brief Signal asynchronous command completion
+ *
+ * Some IPM backends have an ability to deliver a command
+ * asynchronously.  The callback will be invoked in interrupt context,
+ * but the message (including the provided data pointer) will stay
+ * "active" and unacknowledged until later code (presumably in thread
+ * mode) calls ipm_complete().
+ *
+ * This function is, obviously, a noop on drivers without async
+ * support.
+ *
+ * @param ipmdev Driver instance pointer.
+ */
+__syscall void ipm_complete(const struct device *ipmdev);
```

Comment:
Can I recommend allowing a pass through void* userdata as well here if that seems sensible, otherwise you are forcing static data to be used and you may wish to pass in a ipc object alongside the callback here without always creating custom callbacks.

Reply:
To be clear: that function pointer is a method, it's implemented by the driver, not the user of the API.

Reply:
Bah, ignore then, I commented too quickly on this